pkg/cloudprovider/identity.go

```diff
@@ -0,0 +1,92 @@
+package cloudprovider
+
+import (
+	"errors"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
+)
+
+// IdentityHolder represents a resource that contains an Identity object
+// This is used to be able to generically intract with multiple resource types (e.g. VirtualMachine and VittualMachineScaleSet)
+// which each contain an identity.
+type IdentityHolder interface {
+	IdentityInfo() IdentityInfo
+	ResetIdentity() IdentityInfo
+}
+
+// IdentityInfo is used to interact with different implementations of Azure compute identities.
+// This is needed because different Azure resource types (e.g. VirtualMachine and VirtualMachineScaleSet)
+// have different identity types.
+// This abstracts those differences.
+type IdentityInfo interface {
+	AppendUserIdentity(id string)
+	RemoveUserIdentity(id string) error
+}
+
+var (
+	errNotAssigned = errors.New("identity is not assigned to the resource")
```

Comment:
Is it convention to add 'err' before errors. is that required ?

Reply:
It is convention. For error names, when defining an error type usually the end of the name has "Error", for variables it they tend to begin with "err"